tests/test_fpp.py

```diff
@@ -1,20 +1,242 @@
+        self.numbers_multiple = [1, 2, 3, 4, 5]
+        self.single_value = [42]
+        self.mixed_types = [1, 2.5, 3, 4.5]
+        self.zeroes = [0, 0, 0, 0]
+
+    def test_rms_size_array(self):
+        result = rms_size(self.numbers_array)
+        expected = 5.422176684690384
+        self.assertAlmostEqual(result, expected)
+
+    def test_rms_size_multiple_args(self):
+        result = rms_size(*self.numbers_multiple)
+        expected = 3.3166247903554
+        self.assertAlmostEqual(result, expected)
+
+    def test_rms_size_single_value(self):
+        result = rms_size(*self.single_value)
+        expected = 42.0
+        self.assertAlmostEqual(result, expected)
+
+    def test_rms_size_mixed_types(self):
+        result = rms_size(*self.mixed_types)
+        expected = 3.020761493398643
+        self.assertAlmostEqual(result, expected)
+
+    def test_rms_size_zeroes(self):
+        result = rms_size(*self.zeroes)
+        expected = 0.0
+        self.assertAlmostEqual(result, expected)
+
+    def tearDown(self):
+        pass
+    
 class TestStd(unittest.TestCase):
 
-	def setUp(self):
-		self.numbers = [-1,0,1]
-		self.df = pd.DataFrame({'col1': self.numbers})
+    def setUp(self):
+        self.numbers = [-1, 0, 1]
+        self.df = pd.DataFrame({'col1': self.numbers})
+
+    def test_sd(self):
+        sd_ = sd(self.numbers)
+        self.assertFalse(sd_ == 1)
+        apply_pd_changes()
+        self.assertTrue(self.df.col1.sd() == sd_)
+        self.assertTrue(sd_ == sd(*self.numbers))
+
+    def tearDown(self):
+        pass
+
+class TestVariance(unittest.TestCase):
+
+    def setUp(self):
+        self.numbers_array = [-1, 0, 1]
+        self.numbers_multiple = [1, 2, 3, 4, 5]
+        self.single_value = [42]
+        self.mixed_types = [1, 2.5, 3, 4.5]
+        self.zeroes = [0, 0, 0, 0]
+
+    def test_variance_array(self):
+        result = var(self.numbers_array)
+        expected = 0.6666666666666666
+        self.assertAlmostEqual(result, expected)
+
+    def test_variance_multiple_args(self):
+        result = var(*self.numbers_multiple)
+        expected = 2.0
+        self.assertAlmostEqual(result, expected)
+
+    def test_variance_single_value(self):
+        result = var(*self.single_value)
+        expected = 0.0
+        self.assertAlmostEqual(result, expected)
+
+    def test_variance_mixed_types(self):
+        result = var(*self.mixed_types)
+        expected = 1.5625
+        self.assertAlmostEqual(result, expected)
+
+    def test_variance_zeroes(self):
+        result = var(*self.zeroes)
+        expected = 0.0
+        self.assertAlmostEqual(result, expected)
+
+    def tearDown(self):
+        pass
+
+class TestSDPlus(unittest.TestCase):
+
+    def setUp(self):
+        self.numbers_array = [-1, 0, 1]
+        self.numbers_multiple = [1, 2, 3, 4, 5]
+        self.single_value = [42]
+        self.mixed_types = [1, 2.5, 3, 4.5]
+        self.zeroes = [0, 0, 0, 0]
```

Comment:
let's add a test for `self.singleton = [1]` where that should give a zerodivision or another error.